tests/integration-tests/utils.py

```diff
@@ -127,6 +128,51 @@ def random_alphanumeric(size=16):
     return "".join(random.choice(string.ascii_lowercase + string.digits) for _ in range(size))
 
 
+def retrieve_metric_data(unique_name, cluster_name, metric_name, period_length_sec, collection_time_min):
+    """Create Boto3 get_metric_data request and output the results"""
+    assert_that(len(unique_name)).is_equal_to(len(metric_name))
+    metric_queries = []
+    for i in range(len(metric_name)):
+        query = {
+            "Id": unique_name[i],
+            "MetricStat": {
+                "Metric": {
+                    "Namespace": "ParallelCluster/Errors/" + cluster_name,
+                    "MetricName": metric_name[i],
+                },
+                "Period": period_length_sec,
+                "Stat": "Sum",
+            },
+        }
+        metric_queries.insert(i, query)
```

Comment:
NIT: Using an index-based for loop + `list.insert(i, ...)` is fine but using a for-each loop with a `list.append` is a bit more readable.

Reply:
For a more functional way of writing this, you could have a local function that creates the metric:

```
def create_metric(unique_name, metric_name):
    return {
            "Id": unique_name,
            "MetricStat": {
                "Metric": {
                    "Namespace": "ParallelCluster/Errors/" + cluster_name,
                    "MetricName": metric_name,
                },
                "Period": period_length_sec,
                "Stat": "Sum",
            },
        }
```

Then you can use a list comprehension to generate the queries:

```
metric_queries = [create_metric(unique_name, metric_name for unique_name, metric_name in zip(unique_names, metric_names)]
```

which will separate the logic of formatting the individual metric from constructing the list.

Reply:
implemented per charles's suggestion, tested and still works, very good tip thanks!

Reply:
Implemented per Charlie's suggestion. Tests rerun and it works. Thanks for the helpful suggestions